pallets/funding/src/functions.rs

```diff
@@ -849,6 +863,34 @@ impl<T: Config> Pallet<T> {
 				token_information.decimals,
 			)?;
 
+			let contribution_token_treasury_account = T::ContributionTreasury::get();
+			let deposit = T::ContributionTokenCurrency::deposit_required(project_id);
+			T::NativeCurrency::release(
+				&HoldReason::FutureDeposit(project_id).into(),
+				&contribution_token_treasury_account,
+				deposit,
+				Precision::BestEffort,
+			)?;
```

Comment:
Should the tokens be "released" (if necessary if we keep the hold) and returned to the issuer on project_failure?

Reply:
You're right